packages/api/lambdas/process-s3-dead-letter-archive.js

```diff
@@ -43,8 +44,12 @@ async function processDeadLetterArchive({
     const deadLetterObjects = listObjectsResponse.Contents;
     const promises = await Promise.allSettled(deadLetterObjects.map(
       async (deadLetterObject) => {
-        const cumulusMessage = await getJsonS3Object(bucket, deadLetterObject.Key);
+        const deadLetterMessage = await getJsonS3Object(bucket, deadLetterObject.Key);
+        const cumulusMessage = await unwrapDeadLetterCumulusMessage(deadLetterMessage);
         try {
+          if (!cumulusMessage.cumulus_meta || !cumulusMessage.payload || !cumulusMessage.meta) {
```

Comment:
Is `cumulusMessage.meta` always required for writing records? I'm not opposed per se, but why the pre-emptive throw versus letting `writeRecordsFunction` run?